index.js

```diff
@@ -37,59 +36,76 @@ class ChartjsNode {
      *
      * @returns {Promise} A promise that will resolve when the chart is completed
      */
-    drawChart (configuration) {
+    drawChart(configuration) {
         // ensure we clean up any existing window if drawChart was called more than once.
         this.destroy();
         return jsdom.envAsync('<html><body><div id="chart-div" style="font-size:12; width:' + this.width + '; height:' + this.height + ';"><canvas id="myChart" width="' + this.width + '" height="' + this.height + '"></canvas>></div></body></html>',
             [])
-        .then(window => {
-            // these are probably not defined but just to be safe store them
-            var windowTemp = global.window;
-            var docTemp = global.document;
-            global.document = window.document;
-            global.window = window;
-            const Chartjs = require('chart.js');
-            if(configuration.options.plugins) {
-                Chartjs.pluginService.register(configuration.options.plugins);
-            }
-            this._window = window;
-            debug('got window');
-            this._disableDynamicChartjsSettings(configuration);
-            this._canvas = BbPromise.promisifyAll(window.document.getElementById('myChart'));
-            this._ctx = this._canvas.getContext('2d');
-            this._chart = new Chartjs(this._ctx, configuration);
-            global.window = windowTemp;
-            global.document = docTemp;
-            return this._chart;
-        });
+            .then(window => {
+                const canvas = require('canvas');
+                const canvasMethods = ['HTMLCanvasElement'];
+
+                // adding window properties to global (only properties that are not already defined).
+                this._windowPropertiesToDestroy = [];
+                Object.keys(window).forEach(property => {
+                    if (typeof global[property] === 'undefined') {
+                        global[property] = window[property];
+                        this._windowPropertiesToDestroy.push(property);
+                    }
+                });
+
+                // adding all window.HTMLCanvasElement methods to global.HTMLCanvasElement
+                canvasMethods.forEach(method =>
+                    global[method] = window[method]
```

Comment:
actually I suppose this is necessary whether it is defined or not. so fine for me to leave the check out.